drivers/sensor/lps22hh/lps22hh_spi.c

```diff
@@ -0,0 +1,136 @@
+/* ST Microelectronics LPS22HH pressure and temperature sensor
+ *
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lps22hh.pdf
+ */
+
+
+#include <string.h>
+#include "lps22hh.h"
+#include <logging/log.h>
+
+#ifdef DT_ST_LPS22HH_BUS_SPI
+
+#define LPS22HH_SPI_READ		(1 << 7)
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+LOG_MODULE_DECLARE(LPS22HH);
+
+static struct spi_config lps22hh_spi_conf = {
+	.frequency = DT_INST_0_ST_LPS22HH_SPI_MAX_FREQUENCY,
+	.operation = (SPI_OP_MODE_MASTER | SPI_MODE_CPOL |
+		      SPI_MODE_CPHA | SPI_WORD_SET(8) | SPI_LINES_SINGLE),
+	.slave     = DT_INST_0_ST_LPS22HH_BASE_ADDRESS,
+	.cs        = NULL,
+};
+
+static int lps22hh_spi_read(struct lps22hh_data *data, u8_t reg_addr,
+			    u8_t *value, u8_t len)
+{
+	struct spi_config *spi_cfg = &lps22hh_spi_conf;
+	u8_t buffer_tx[2] = { reg_addr | LPS22HH_SPI_READ, 0 };
+	const struct spi_buf tx_buf = {
+			.buf = buffer_tx,
+			.len = 2,
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+	const struct spi_buf rx_buf[2] = {
+		{
+			.buf = NULL,
+			.len = 1,
+		},
+		{
+			.buf = value,
+			.len = len,
+		}
+	};
+	const struct spi_buf_set rx = {
+		.buffers = rx_buf,
+		.count = 2
+	};
+
+
+	if (len > 64) {
+		return -EIO;
+	}
+
+	if (spi_transceive(data->bus, spi_cfg, &tx, &rx)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int lps22hh_spi_write(struct lps22hh_data *data, u8_t reg_addr,
+			     u8_t *value, u8_t len)
+{
+	struct spi_config *spi_cfg = &lps22hh_spi_conf;
+	u8_t buffer_tx[1] = { reg_addr & ~LPS22HH_SPI_READ };
+	const struct spi_buf tx_buf[2] = {
+		{
+			.buf = buffer_tx,
+			.len = 1,
+		},
+		{
+			.buf = value,
+			.len = len,
+		}
+	};
+	const struct spi_buf_set tx = {
+		.buffers = tx_buf,
+		.count = 2
+	};
+
+
+	if (len > 64) {
+		return -EIO;
+	}
+
+	if (spi_write(data->bus, spi_cfg, &tx)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+lps22hh_ctx_t lps22hh_spi_ctx = {
```

Comment:
Why isn't this part of `struct lps22hh_data`?

Reply:
moved in data